pkg/events/events.go

```diff
@@ -168,7 +168,7 @@ func Proto(e Event) (*ttnpb.Event, error) {
 			pb.Data, err = types.MarshalAny(protoMessage)
 		} else if err, ok := evt.data.(error); ok {
 			if ttnErr, ok := errors.From(err); ok {
-				pb.Data, err = types.MarshalAny(ttnErr.GRPCStatus().Proto())
+				pb.Data, err = types.MarshalAny(ttnpb.ErrorDetailsToProto(ttnErr))
```

Comment:
This still sets the `err` that is locally redefined on L169. So either check-return this `err` on L175 or rename `err` on L169

Reply:
Note that this is also why we saw a half message printed; the error wasn't caught but we used a dirty buffer